pkg/secrets/kubernetes_secrets.go

```diff
@@ -2,55 +2,130 @@ package secrets
 
 import (
 	"fmt"
-
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/utils"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
-func storeSecretsInKubeSecrets(namespace string, secretName string, secrets []Secret) error {
-
-	// Create a map of secret key to value to place in the
-	// Kubernetes secret
-	secretsData := make(map[string][]byte)
-	for _, secret := range secrets {
-		secretsData[secret.Key] = secret.SecretBytes
-	}
+type K8sSecret struct {
+	Secret *v1.Secret
+}
 
+func retrieveK8sSecret(namespace string, secretName string) (*K8sSecret, error) {
+	// TODO: extract this to another function
 	// Create the Kubernetes client
 	InfoLogger.Print("Creating Kubernetes client...")
 	kubeConfig, err := rest.InClusterConfig()
 	if err != nil {
-		return fmt.Errorf("Failed to load in-cluster Kubernetes client config: %s", err)
+		return nil, fmt.Errorf("failed to load in-cluster Kubernetes client config: %s", err)
 	}
 
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
-		return fmt.Errorf("Failed to create Kubernetes client: %s", err)
+		return nil, fmt.Errorf("failed to create Kubernetes client: %s", err)
 	}
 
-	// Create the Kubernetes secret
-	secret := v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: secretName,
-		},
-		Data: secretsData,
+	// TODO: extract until here
+
+	InfoLogger.Printf("Retrieving Kubernetes secret '%s' from namespace '%s'...", secretName, namespace)
+	k8sSecret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
```

Comment:
Here we are assuming that all of customer's Conjur secrets will be wrapped up as data entries in a single K8s secret?

Reply:
we don't. but each k8s secret (that needs secrets from conjur) will have a "conjur-map" entry which will hold the required secrets *for that k8s secret*. we may have more than one k8s secret so we run the operation above for each one